pandas/io/formats/format.py

```diff
@@ -1513,20 +1588,30 @@ def _format_strings(self):
 
 
 class Timedelta64Formatter(GenericArrayFormatter):
-    def __init__(self, values, nat_rep="NaT", box=False, **kwargs):
+    def __init__(
+        self,
+        values: Union[ndarray, TimedeltaIndex],
```

Comment:
really?

Reply:
so due the fact that `_format_native_types` in `pandas/core/indexes/timedeltas.py` uses `Timedelta64Formatter` directly, this should account for the `TimedeltaIndex`.